### ISSUE TITLE
[BUG] oiiotool with -i:ch flag crashes on multipart exr

### ISSUE THREAD

**Issue by naniBox** [OpenImageIO/oiio#3509]
**Describe the bug**
Using oiiotool to read an EXR with a `<z_channel>` while using the flag `-i:ch=R,G,B,A` results in a crash. If changing the flag to `-i <file> --ch R,G,B,A`, it does not crash. If the file does not have the `<z_channel>`, it does not crash.

The original EXR was created from a 3delight render, with an extra Image Layer (AOV) "Z (depth)".

The crash is happening with OIIO `2.3.10`, `2.1.9`, `2.4.0dev` (from Arnold). The command silently fails (no crash) with OIIO from Arnold `2.2.1`, `2.1.4dev`. The command works as expected with OIIO from Arnold `2.1.0dev`. All my tests were done on Windows10.

**To Reproduce**
1. Call `oiiotool.exe -i:ch=R,G,B,A "in.exr" -o "out.exr"`
2. Crash:
``` 0# OpenImageIO_v2_3::Sysutil::hardware_concurrency in OpenImageIO_Util
 1# OpenImageIO_v2_3::Sysutil::stacktrace in OpenImageIO_Util
 2# OpenImageIO_v2_3::Sysutil::stacktrace in OpenImageIO_Util
 3# seh_filter_exe in ucrtbase
 4# 0x00007FF66322FC4A in oiiotool
 5# _C_specific_handler in VCRUNTIME140
 6# _chkstk in ntdll
 7# RtlRaiseException in ntdll
 8# KiUserExceptionDispatcher in ntdll
 9# OpenImageIO_v2_3::ImageBuf::reset in OpenImageIO
10# OpenImageIO_v2_3::ImageBufAlgo::channels in OpenImageIO
11# 0x00007FF6631B4748 in oiiotool
12# 0x00007FF66320CCA7 in oiiotool
13# 0x00007FF6632032B9 in oiiotool
14# OpenImageIO_v2_3::ArgParse::parse_args in OpenImageIO_Util
15# OpenImageIO_v2_3::ArgParse::parse_args in OpenImageIO_Util
16# 0x00007FF6631FEF5D in oiiotool
17# 0x00007FF663216331 in oiiotool
18# 0x00007FF66322ADA0 in oiiotool
19# BaseThreadInitThunk in KERNEL32
20# RtlUserThreadStart in ntdll
 0# OpenImageIO_v2_3::Sysutil::hardware_concurrency in OpenImageIO_Util
 1# OpenImageIO_v2_3::Sysutil::stacktrace in OpenImageIO_Util
 2# OpenImageIO_v2_3::Sysutil::stacktrace in OpenImageIO_Util
 3# raise in ucrtbase
 4# OpenImageIO_v2_3::Sysutil::stacktrace in OpenImageIO_Util
 5# seh_filter_exe in ucrtbase
 6# 0x00007FF66322FC4A in oiiotool
 7# _C_specific_handler in VCRUNTIME140
 8# _chkstk in ntdll
 9# RtlRaiseException in ntdll
10# KiUserExceptionDispatcher in ntdll
11# OpenImageIO_v2_3::ImageBuf::reset in OpenImageIO
12# OpenImageIO_v2_3::ImageBufAlgo::channels in OpenImageIO
13# 0x00007FF6631B4748 in oiiotool
14# 0x00007FF66320CCA7 in oiiotool
15# 0x00007FF6632032B9 in oiiotool
16# OpenImageIO_v2_3::ArgParse::parse_args in OpenImageIO_Util
17# OpenImageIO_v2_3::ArgParse::parse_args in OpenImageIO_Util
18# 0x00007FF6631FEF5D in oiiotool
19# 0x00007FF663216331 in oiiotool
20# 0x00007FF66322ADA0 in oiiotool
21# BaseThreadInitThunk in KERNEL32
22# RtlUserThreadStart in ntdll
```

**Expected behavior**
A generated `out.exr` with RGBA layers only.

**Evidence**
Attached [in.zip](https://github.com/OpenImageIO/oiio/files/9256249/in.zip).

**Platform information:**
- Input formats supported: bmp, cineon, dds, dpx, ffmpeg, fits, gif, hdr, iff, jpeg, null, openexr, png, pnm, psd, raw, rla, sgi, socket, softimage, targa, tiff, webp, zfile
- Output formats supported: bmp, dpx, fits, gif, hdr, iff, jpeg, null, openexr, png, pnm, rla, sgi, socket, targa, term, tiff, webp, zfile
- OpenColorIO 2.1.1, color config: built-in
- Known color spaces: "linear", "default", "rgb", "RGB", "sRGB", "Rec709"
- Filters available: box, triangle, gaussian, sharp-gaussian, catmull-rom, blackman-harris, sinc, lanczos3, radial-lanczos3, nuke-lanczos6, mitchell, bspline, disk, cubic, keys, simon, rifman
- Dependent libraries: FFMpeg 4.4.1 (Lavf58.76.100), gif_lib 5.2.1, jpeg-turbo 2.1.3/jp62, null 1.0, OpenEXR 2.5.0, libpng 1.6.37, libraw 0.19.0-Beta1, LIBTIFF Version 4.3.0, Webp 1.2.2
- OIIO 2.3.10 built for C++14/199711 sse2
- Running on 8 cores 31.9GB sse2,sse3,ssse3,sse41,sse42,avx,avx2,fma,f16c,popcnt,rdrand
Windows 10 Pro 64

**Comment by lgritz** [comment]
I can reproduce with this example, thanks. Looking into it, stay tuned...

**Comment by lgritz** [comment]
Proposed fix in #3513

If you need a workaround, it should work to just use `--ch` separately from -i:

    oiiotool in.exr --ch  R,G,B,A -o out.exr
